Cosmostation/Controller/Main/MainTabWalletViewController.swift

```diff
@@ -856,10 +841,20 @@ class MainTabWalletViewController: BaseViewController, UITableViewDelegate, UITa
         }
     }
     
-    func onStartMoonPay(_ url: String) {
-        let urlMoonpay = URL(string: url)
-        if(UIApplication.shared.canOpenURL(urlMoonpay!)) {
-            UIApplication.shared.open(urlMoonpay!, options: [:], completionHandler: nil)
+    func onShowBuyWarnNoKey() {
+        let noKeyAlert = UIAlertController(title: NSLocalizedString("buy_without_key_title", comment: ""),
+                                           message: NSLocalizedString("buy_without_key_msg", comment: ""),
+                                           preferredStyle: .alert)
+        noKeyAlert.overrideUserInterfaceStyle = BaseData.instance.getThemeType()
+        noKeyAlert.addAction(UIAlertAction(title: NSLocalizedString("cancel", comment: ""), style: .default, handler: {_ in
+            self.dismiss(animated: true, completion: nil)
+        }))
+        noKeyAlert.addAction(UIAlertAction(title: NSLocalizedString("continue", comment: ""), style: .destructive, handler: {_ in
+            self.onCheckCryptoPay()
+        }))
+        self.present(noKeyAlert, animated: true) {
+            let tapGesture = UITapGestureRecognizer(target: self, action: #selector(self.dismissAlertController))
```

Comment:
not recommended pattern on iOS
if Apple offers a way to handle this interaction is better to follow the best practices, otherwise if the APIs change this code can break or worse, make the app crash as it is accessing directly a subviews array assuming that it will contain a UIView in the position [0], if not the app will crash